code/game/machinery/washing_machine.dm

```diff
@@ -14,306 +13,401 @@
+	. = ..()
+	if(washing)
+		return
+	if(!door_open)
+		toggle_door()
+		return
 	else
-		return ..()
+		L.forceMove(loc)
+		check_tub_contents()
 
-/obj/machinery/washing_machine/attack_hand(mob/user)
-	switch(state)
-		if(OPEN_EMPTY)
-			state = CLOSED_EMPTY
-		if(CLOSED_EMPTY)
-			for(var/atom/movable/O in src)
-				O.forceMove(loc)
-			crayon = null
-			state = OPEN_EMPTY
-		if(OPEN_FULL)
-			state = CLOSED_FULL
-		if(CLOSED_FULL)
-			for(var/atom/movable/O in src)
-				O.forceMove(loc)
-			crayon = null
-			state = OPEN_EMPTY
-		if(RUNNING)
-			to_chat(user, "<span class='warning'>[src] is busy.</span>")
-		if(CLOSED_BLOODY)
-			if(gibs_ready)
-				gibs_ready = FALSE
-				if(locate(/mob) in src)
-					var/mob/M = locate() in src
-					M.gib()
-			for(var/atom/movable/O in src)
-				O.forceMove(loc)
-			crayon = null
-			state = OPEN_EMPTY
-
-	update_icon(UPDATE_ICON_STATE)
+
+
+/obj/machinery/washing_machine/default_deconstruction_screwdriver(mob/user, icon_state_open, icon_state_closed, obj/item/I)
+	if(I.tool_behaviour != TOOL_SCREWDRIVER)
+		return FALSE
+	if(!I.use_tool(src, user, 0, volume = 0))
+		return FALSE
+	if(!panel_open)
+		panel_open = TRUE
+		to_chat(user, "<span class='notice'>You open the maintenance hatch of [src].</span>")
+	else
+		panel_open = FALSE
+		to_chat(user, "<span class='notice'>You close the maintenance hatch of [src].</span>")
+		I.play_tool_sound(user, I.tool_volume)
+	update_icon_state(UPDATE_ICON_STATE)
+	return TRUE
 
 /obj/machinery/washing_machine/deconstruct(disassembled = TRUE)
-	new /obj/item/stack/sheet/metal(drop_location(), 2)
+	eject_tub_contents()
+	new /obj/item/stack/sheet/metal(drop_location(), 5)
+	new /obj/item/circuitboard/washing_machine(drop_location(), 2)
+	qdel(src)
+
+/obj/machinery/washing_machine/AllowDrop()
+	return TRUE // so gibs and dropped items end up
+
+/obj/machinery/washing_machine/cleaning_act(mob/user, atom/cleaner, cleanspeed, text_verb, text_description, text_targetname)
+	bloody_mess = FALSE
+	return ..()
+
+/obj/item/storage/belt/fannypack/machine_wash(obj/machinery/washing_machine/washer)
+	. = ..()
+	if(washer.color_source)
+		dye_item(washer.wash_color)
+
+/obj/item/clothing/machine_wash(obj/machinery/washing_machine/washer)
+	. = ..()
+	if(washer.color_source)
+		dye_item(washer.wash_color)
+
+/obj/item/bedsheet/machine_wash(obj/machinery/washing_machine/washer)
+	. = ..()
+	if(washer.color_source)
+		dye_item(washer.wash_color)
+
+/obj/item/holder/machine_wash(obj/machinery/washing_machine/washer)
+	. = ..()
+	for(var/mob/living/held_mob in contents)
+		held_mob.machine_wash(washer)
+
+/obj/item/stack/sheet/hairlesshide/machine_wash(obj/machinery/washing_machine/washer)
+	. = ..()
+	new /obj/item/stack/sheet/wetleather(loc, amount)
 	qdel(src)
 
-#undef OPEN_EMPTY
-#undef CLOSED_EMPTY
-#undef OPEN_FULL
-#undef CLOSED_FULL
-#undef RUNNING
-#undef CLOSED_BLOODY
-#undef RUNNING_BLOODY
+/mob/living/simple_animal/pet/machine_wash(obj/machinery/washing_machine/washer)
```

Comment:
This means that `/mob/living/simple_animal/mouse` won't be gibbed in the washer, but they can be put in because of /obj/item/holder

Reply:
Yep. I can expand the content if you want. You can actually put most simple_animals in anyway given they have a small enough `mob_size` I just haven't done any special behaviour on the simple_animal type due to me worrying about accidently allowing people to gib certain simple mobs with ease.

Reply:
just expanded this to all simple mobs with Small size or lower